src/facade/dragonfly_connection.h

```diff
@@ -105,6 +94,24 @@ class Connection : public util::Connection {
     StorageType storage;
   };
 
+  // Monitor message, carries a simple payload with the registered event to be sent.
+  struct MonitorMessage : public std::string {};
+
+  // ACL Update message, contains ACL updates to be applied to the connection.
+  struct AclUpdateMessage {
+    std::vector<std::string> username;
+    std::vector<uint32_t> categories;
+    std::vector<std::vector<uint64_t>> commands;
+  };
+
+  // Migration request message, the dispatch fiber stops to give way for thread migration.
+  struct MigrationRequestMessage {};
+
+  // Checkpoint message, decrements counter when processed.
```

Comment:
please expand when (how?) checkpoint message is used.

Reply:
i.e. "dummy message used to make sure that all the currently queues messages are processed in the queue"